contextual_orchestrator/model_discovery.py

```
@@ -0,0 +1,367 @@
+
+def _fetch_json(url: str, *, api_key: str, auth_scheme: str, timeout: float) -> Any:
+    parsed = urlparse(url)
+    if parsed.scheme in {"local", "mlx"}:
+        if parsed.hostname not in _LOCAL_DISCOVERY_HOSTS:
+            raise ValueError("local model discovery requires a loopback provider host")
+        url = urlunparse(("http", parsed.netloc, parsed.path, "", "", ""))
+    request = urllib.request.Request(
+        url,
+        headers={"authorization": f"{auth_scheme} {api_key}"},
+        method="GET",
+    )
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)